torch/csrc/jit/codegen/cuda/index_compute.cpp

```diff
@@ -773,7 +805,8 @@ generateIndexAndExtentMap(
 kir::TensorIndex* Index::getGlobalProducerIndex(
     TensorView* producer_tv,
     const TensorView* consumer_tv,
-    const std::vector<kir::ForLoop*>& loops) {
+    const std::vector<kir::ForLoop*>& loops,
```

Comment:
same thing here: we're passing too much stuff as arguments

Reply:
Just wondering, would it make sense to make `Index` stateful instead of being a pure interface? It can have `ca_root_map` as a private member so it doesn't need to be carried around in all the API calls.

Reply:
That sounds like a good idea. Thanks! 

I think that index_compute.cpp could use some refactoring. In this PR, though, I'd keep the code change as minimum as possible since its logical change is not trivial.